### PR TITLE
Add full iOS job in package pipeline

### PR DESCRIPTION
**Pull request by gwang-msft**
**Description**: Add full Android job in package pipeline

**Motivation and Context**
- Nuget/Xamarin package need onnxruntime full instead of ORT-Moible, add a iOS full xcframework job in the packaging pipeline
- Move the current iOS CI to use iOS 11.0 as target
- `std::optional` does not have `value()` until iOS 12, change the usage from `std::optional::value()` to `*std::optional` in relevant code
- Test run, https://aiinfra.visualstudio.com/Lotus/_build/results?buildId=174116&view=logs&j=511b6d5b-a43c-51a8-3008-242664c9ff2d

### CODE REVIEW

**Review by snnn — APPROVED** [review]

**Review by skottmckay — APPROVED** [review]
:shipit:

File: tools/ci_build/github/azure-pipelines/templates/c-api-cpu.yml [hunk 1]
```diff
@@ -183,6 +183,36 @@ jobs:
...
+  timeoutInMinutes:  180
+  steps:
+    - script: |
+        set -e -x
+        python3 tools/ci_build/github/apple/build_ios_framework.py \
+          --build_dir "$(Build.BinariesDirectory)/ios_framework" \
+          tools/ci_build/github/apple/default_full_ios_framework_build_settings.json
+        mkdir $(Build.BinariesDirectory)/artifacts
+        pushd $(Build.BinariesDirectory)/ios_framework/framework_out/
+        zip -vr $(Build.BinariesDirectory)/artifacts/onnxruntime_xcframework.zip onnxruntime.xcframework
```
Comment (skottmckay) [inline tools/ci_build/github/azure-pipelines/templates/c-api-cpu.yml:200]: nit: is it possible to add a top level folder to the zip called something like onnxruntime-ios-xcframework? the extraction script that sets things up for insertion into the nuget package expects a directory containing the binaries with a trailing '-xyz' that it trims off.

https://github.com/microsoft/onnxruntime/blob/e83cc534d4bb8e7c76433adad010cce3cc6fc9d8/tools/ci_build/github/windows/extract_nuget_files.ps1#L39

Comment (gwang-msft) [reply tools/ci_build/github/azure-pipelines/templates/c-api-cpu.yml:200]: Can this be done in the download artifacts step? You can add whatever folder structure you want in the step

Comment (gwang-msft) [reply tools/ci_build/github/azure-pipelines/templates/c-api-cpu.yml:200]: Will address this in a follow up PR

**Pull request merged**